---
title: "this is for the result part of the draft"
author: "Wei Wang"
date: 2017-03-15
output: html_document
---

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
```{r read-chunk, include=FALSE, cache=FALSE}
knitr::read_chunk("chunks.R")
```

<!-- Update knitr chunk options -->
```{r knitr-opts-chunk, include=FALSE}
```

<!-- Insert the date the file was last updated -->
```{r last-updated, echo=FALSE, results='asis'}
```

<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
```{r code-version, echo=FALSE, results='asis'}
```

<!-- Add your analysis here -->

## data analysis on the normalized data

### missing data imputation

#### Tumors data

We test the performance of our method on the real data where there is no ground truth. The performance is assessed by the predictive Mean Square Error (MSE) for the missing value, where the true value is know and chosen to be missing. This data set is a microarray data from 'denoiseR' package containing 43 brain tumors of four types and 356 continous variable related with the expression data. we set the rank equal to 20 for FLASH, PMD and softImpute, since we find that the rank of the low rank structure is almost always smaller than 20, whihc is around 15-20.

![tumor data](~/HG/ash-sfa/Rcode/postmean/flash_simulation/denoise/denoiseRdata/tumordata/boxplot.png)

#### Movie data 

The MovieLens data set is available at the GroupLens Research group which is usualy used for the exploration of recommender systems. The idea of the prediction is that users with similar preference on movies might have similar ratings so that we can predict tha ratings of certain user based on other users sharing some common preference. There are several data sets available for stable bench mark and we choose the smallest one. The MovieLens 100K data set we used here is a data set with about 100K ratings from 1000 user and 1700 movies. The ratings are integers from 0 to 5 represent preference of users. In this data, all the ratings are nonnegative, so we center and scale the data for each users. We focus on the structure that the users share common prefernce on movies, so the preference of each user on each movies remains the same.

![Movie data](~/HG/ash-sfa/Rcode/postmean/flash_simulation/missingvalue/ML100K/boxplotrowscale.png)

#### presidential address

This data set is a contingency table cross-tabulating with 13 rows and 836 columns, where rows stands for US presidents from 1940 to 2009 and columns are correspoding the words they used in thier inaugural addresses. This data set comes from 'denoiseR' R package. We make one more pre-processing procedure before apply all the methods. The pre-precessed data is center and scale on both rows ans columns. It is because the words and the frequency of those words vary a lot among different presidents. This is data is different from movie ratings data of which the ratings can only take values from 0 to 5. Some words show up more frequently than others, such as "a" and "the", so we should center and scale the data not only by rows but also by coumns.

![president data](~/HG/ash-sfa/Rcode/postmean/flash_simulation/denoise/denoiseRdata/presidentdata/biscaleboxplot.png)

#### GTEx eQTL summary statistic

This data set is summary statistics (Z-score) from eQTL analysis.

![GTEx data](~/HG/ash-sfa/Rcode/postmean/flash_simulation/missingvalue/GTEXboxplot.png)

This data set is 

#### Breast cancer data

The Breast cancer data is from West et al. (2007). There are 251 samples and 226 genes in this data set. For those algorithm we choose, we set the rank equal to 50 which is close the the rank of the estimation from the NSFA results. After some try of FLASH, we find that the rank picked by FLASH is also around 40-50. Following the pre-processing procedure from NSFA, we also use the centered data. 

![Breast Cancer data](~/HG/ash-sfa/Rcode/postmean/flash_simulation/missingvalue/Breastcancer/boxplot_centered.png)



### gene expression data

To show the flexibility of FLASH, we take the summary statistics from the expression Quantitative Trait Loci (eQTLs) of 44 human Tissues. The original data is from Genotype Tissue Expression
(GTEx) project. The aim of GTEx project "s to increase our understanding of how changes in our genes contribute to common human diseases, in order to improve health care for future generations."
We try to understand the " which SNPs are eQTLs (i.e. associated with expression) in which tissues.Answering this could help distinguish regulatory regions and mechanisms that are specific to a few tissues vs shared among many tissues.It could also help with analyses that aim to integrate eQTL results with GWAS results to help identify the tissues that are most relevant to any specific complex disease (e.g. [18, 19])."

We apply FLASH on the matrix of summary statistics which are Z-scores. 

"We analysed the 16,069 genes for which univariate effect estimates were available for all 44 tissues we considered; the filtering criteria used ensure that these genes show at least some indication of expression in all 44 tissues." 


As input to mash we use a matrix of eQTL effect estimates ˆbjr, and
corresponding standard errors ˆsij , where the rows j index different SNP-gene
pairs and the columns r index tissues (or cell types). We used the effect
estimates and standard errors for candidate local (“cis”) eQTLs for each gene,
distributed by the GTEx project (v6 release). These were obtained by
(univariate) single-SNP analyses in each tissue by applying MatrixEQTL [20] on
expression levels that have been rank-transformed to the corresponding
quantiles of a standard normal distribution. Thus the effect size estimates are in
units of standard deviations on this transformed scale. Because, like most eQTL
analyses, these estimates were obtained by single-SNP analysis, the estimated
effects for each SNP actually reflect the effects of both the SNP itself and other
SNPs in LD with it (see Supplementary Text for discussion). We analysed the
16,069 genes for which univariate effect estimates were available for all 44 tissues
we considered; the filtering criteria used ensure that these genes show at least
some indication of expression in all 44 tissues.

## other potential extension

### NMF

In ashr, there is a option that we can choose the prior to be half uniform or half normal, which leads to the final estimate is non-negative. In this case, we apply FLASH for non-negative matrix factorization with proper intial values. For example, we can use NMF result as intial value and apply FLASH with backfitting algorithm to get a flexible sparse solutions of factors and loadings.

### tensor flash

In our frame work, it is easy to extend out method to tensor factorization. And the benfit of FLASH is we don't need to estimate the tuning parmeters in each mode.

### rank decision issue

there is somethig we don't fully understand.

## Session information

<!-- Insert the session information into the document -->
```{r session-info}
```
